SpotifyAPI.Web/SpotifyWebAPI.cs

```diff
@@ -2434,6 +2439,21 @@ private static ListResponse<T> ExtractDataToListResponse<T>(Tuple<ResponseInfo,
             return response.Item2;
         }
 
+        private bool TryGetTooManyRequests(ResponseInfo info, out int secondsToWait)
+        {
+            if (info.SpotifyStatusCode != SpotifyHttpStatusCode.TooManyRequests)
+            {
+                secondsToWait = -1;
+                return false;
+            }
+            if (!int.TryParse(info.Headers.Get("Retry-After"), out secondsToWait))
+            {
+                secondsToWait = -1;
+                return false;
+            }
+            return true;
+        }
+
```

Comment:
I want to avoid `out` as much as possible, maybe change return type to `int` and return `-1` on failure?

Reply:
I personally am huge fan of the TryGet pattern
1)  It's absolutely clear what a success is.  I don't have to check comments to know whether I need to test if the return value > 0 or != -1, or if it's going to throw an exception as some parse/get functions do, etc.  Return bool is as straightforward as it can get
2)  Removes the need for a local variable, which helps nest if statements and shortcircuit properly:

    if (item1.SomeTestThing()
       && TryGetTooManyRequests(item1, out var seconds1))
    {
        await Task.Delay(seconds1 * 1000);
        // Other stuff
    }

That's nice because i dont actually run the TryGetTooManyRequests unless SomeTestThing() passes.  I'd have to write/indent much more to get an equivalent statement otherwise:

    if (item1.SomeTestThing())
    {
        var secondsToWait = GetTooManyRequests(item1);
        if (secondsToWait > 0)
        {
            await Task.Delay(seconds1 * 1000);
            // Other stuff
        }
    }

We could perhaps offer both to the user?  A GetTooManyRequests function alongside a TryGetTooManyRequests?  This would be similar to the int.Parse() and int.TryParse() choice, for example.  Definitely just a preference thing, though.  I can keep it simple and change the pattern to match the rest of the library, too.  Just voicing my two cents 8)